File: ch5/clip1/boj10844/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        final int MOD = 1000000000;
        int n = sc.nextInt();
        int[][] d = new int[n + 1][10];
        for (int i = 1; i <= 9; i++) {
            d[1][i] = 1;
        }
        for (int i = 2; i <= n; i++) {
            for(int j = 0; j <= 9; j++) {
                switch (j) {
                    case 0 -> d[i][j] = d[i - 1][j + 1] % MOD;
                    case 9 -> d[i][j] = d[i - 1][j - 1] % MOD;
                    default -> d[i][j] = (d[i - 1][j - 1] + d[i - 1][j + 1]) % MOD;
                }
            }
        }
        int sum = 0;
        for (int i = 0; i <= 9; i++) {
            sum = (sum + d[n][i]) % MOD;
        }
        System.out.println(sum);
    }
}

File: ch5/clip1/boj10844/main.py
MOD = 1000000000
n = int(input())
d = [[0] * 10 for _ in range(n + 1)]

for i in range(1, 10):
    d[1][i] = 1

for i in range(2, n + 1):
    for j in range(10):
        if j == 0:
            d[i][j] = d[i - 1][j + 1] % MOD
        elif j == 9:
            d[i][j] = d[i - 1][j - 1] % MOD
        else:
            d[i][j] = (d[i - 1][j - 1] + d[i - 1][j + 1]) % MOD

sum_value = sum(d[n]) % MOD
print(sum_value)
